### ISSUE TITLE
Docker with python 3.6

### ISSUE THREAD

**Issue by SumNeuron** [tensorflow/tensorflow#22292]
Please update your [nvidia docker file](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/dockerfiles/dockerfiles/nvidia.Dockerfile) to support python 3.6.

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: tried to take relevant parts from docker file, but I get timeouts.
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 16.04 (yes has weird python 3.5.3 dependency, but doesn't mean you can't make python3.6 docker image)
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**: N/a
- **TensorFlow installed from (source or binary)**: trying to use tensorflow-gpu>=1.10
- **TensorFlow version (use command below)**: 1.10+
- **Python version**: 3.6+ 
- **Bazel version (if compiling from source)**: N/a
- **GCC/Compiler version (if compiling from source)**: N/a
- **CUDA/cuDNN version**: nvidia/cuda:9.0-base-ubuntu16.04
- **GPU model and memory**: Titan
- **Exact command to reproduce**:

```
FROM nvidia/cuda:9.0-base-ubuntu16.04
FROM python:3.6-slim
RUN apt-get update && apt-get install \
        cuda-command-line-tools-9-0 \
        libcudnn7=7.2.1.38-1+cuda9.0 \
        libnccl2=2.2.13-1+cuda9.0
...
RUN pip install tensorflow-gpu==1.10.0
```

### Describe the problem
Describe the problem clearly here. Be sure to convey here why it's a bug in TensorFlow or a feature request.

Python 3.6 introduces some non-backwards compatible features - most notably f-strings. Any code with an f-string anywhere will not work with any existing official nvidia or tensorflow docker image

### Source code / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.

**Comment by SumNeuron** [comment]
To clarify, I mean that taking the NVIDIA partial of the TF docker file:

```
# Pick up some TF dependencies                       

RUN apt-get update && apt-get install -y --no-install-recommends \                       

        build-essential \                       

        cuda-command-line-tools-9-0 \                       

        cuda-cublas-9-0 \                       

        cuda-cufft-9-0 \                       

        cuda-curand-9-0 \                       

        cuda-cusolver-9-0 \                       

        cuda-cusparse-9-0 \                       

        libcudnn7=7.2.1.38-1+cuda9.0 \                       

        libnccl2=2.2.13-1+cuda9.0 \                       

        libfreetype6-dev \                       

        libhdf5-serial-dev \                       

        libpng12-dev \                       

        libzmq3-dev \                       

        pkg-config \                       

        software-properties-common \                       

        unzip \                       

        && \                       

    apt-get clean && \                       

    rm -rf /var/lib/apt/lists/*                       

RUN apt-get update && \                       

        apt-get install nvinfer-runtime-trt-repo-ubuntu1604-4.0.1-ga-cuda9.0 && \                       

        apt-get update && \                       

        apt-get install libnvinfer4=4.1.2-1+cuda9.0
```

results in errors (unable to locate package), timeout or the like. So even if I wanted to make the python3.6 version myself, I can not :) 

Trying to use `FROM tensorflow/tensorflow` and `python3` will not work without a virtual environment... as non docker guru, I think many in the community would appreciate it

@NegatioN that is likely to happen, but many people will still need 16.04LTS + py3.6

**Comment by gqoew** [comment]
+1 for a re-buildable, customisable Dockerfile with:

- cuda
- tf-deps
- python 3.6.

**Comment by carlthome** [comment]
Same issue here!

**Comment by SumNeuron** [comment]
@agrinh it is possible with ubuntu 18.04, but again, there will still be a fraction of the community requiring ubuntu 16.04.

either way, I think we would all appreciate if @angersson  could give us an update

**Comment by hyzhak** [comment]
btw until we wait `3.7.1` is already there ;) ⏳ . Could we skip 3.6 and migrate to the last one?

**Comment by SumNeuron** [comment]
@angersson can we please get a "working on it" or an eta....

**Comment by tensorflowbutler** [comment]
Nagging Assignee @angersson: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by maystroh** [comment]
Did anything come up with this ? @angersson

**Comment by angersson** [comment]
@maystroh See https://github.com/tensorflow/tensorflow/pull/24051. Also, I think the nightly CPU images are based on Ubuntu 18.04 now, so those should have a newer Python version.

**Comment by maystroh** [comment]
Thanks @angersson

**Comment by gqoew** [comment]
`tensorflow/tensorflow:nightly-gpu-py3` has Cuda 10.0 but Python 3.5.2.

How to get an image with Cuda 10.0 and Python >= 3.6?

**Comment by desprit** [comment]
Python3.6 offers many cool features, it would be great to have it by default instead of 3.5.

**Comment by jingw222** [comment]
Seriously, the image tag should explicitly specify Python version up to one decimal point. This has got me lots of headaches when there's an underlying Python version discrepancy between testing and deployment.

**Comment by angersson** [comment]
#28717 will upgrade our future Docker images to Ubuntu 18.04. TF 1.14's images should have 18.04 as a base once they're available. 

@jingw222 Good idea -- that may be something we can do later if it turns out to be simple enough. For now, I'd recommend deriving a custom image from ours if you require specific component versions.

**Comment by minimaxir** [comment]
This is *not* fixed: the new images indeed now indeed use Ubuntu 18.04, but the installed Python is still Python 3.5.2 (checking `latest-py3`).

**Comment by angersson** [comment]
@minimaxir, that is correct. Starting from TF 1.14 our images will be based on Ubuntu 18.04. Since 1.14's release process has been delayed, the `latest` tags still point to the `1.13`-tagged images based on Ubuntu 16.04. The `nightly` tags have jumped into 18.04 with Python 3.6:

```
$ docker run -it --rm tensorflow/tensorflow:nightly-py3 python3 -V 
Python 3.6.7
```

**Comment by taneta** [comment]
@angersson Good news!
How about gpu version?
`$ docker run -it --rm tensorflow/tensorflow:nightly-gpu-py3 python3 -V `
`Python 3.5.2`

**Comment by patientzero** [comment]
I created an image that uses python 3.6 and has tensorflow 2.0a-gpu installed. 

`$ docker pull patientzero/tensorflow2.0a-gpu-py3.6`

The repo is here: https://github.com/patientzero/tensorflow2.0a-python3.6-Docker

**Comment by angersson** [comment]
@patientzero Thanks! Please note also that our official `2.0.0b0` suite of images (e.g. `tensorflow/tensorflow:2.0.0b0-gpu-py3`), published this morning with the release of the 2.0 beta, also contain Python 3.6.
